cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/common/DataprocImageVersion.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.runtime.spi.common;
+
+import io.cdap.cdap.runtime.spi.VersionInfo;
+import joptsimple.internal.Strings;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+
+/**
+ * Class used to compare versions from Dataproc cluster images
+ */
+public class DataprocImageVersion implements VersionInfo {
+  List<Integer> versionSegments;
+
+  public DataprocImageVersion(@Nullable String imageVersion) {
+    // For null versions, we assume zero.
+    if (imageVersion == null) {
+      imageVersion = "0";
+    }
+
+    versionSegments = new ArrayList<>();
+
+    // If the version contains a hyphen, only take the first portion (e.g. 1.2.3-debian4)
+    if (imageVersion.contains("-")) {
+      imageVersion = imageVersion.split("-")[0];
+    }
+
+    // As a safety, assume that missing elements default to zero. (e.g. 1.3. -> 1.3.0, 1..1 -> 1.0.1)
+    for (String component : imageVersion.split("\\.")) {
+      Integer part = Strings.isNullOrEmpty(component) ? 0 : Integer.parseInt(component);
+      versionSegments.add(part);
+    }
+  }
+
+  @Override
+  public int compareTo(Object o) {
+    DataprocImageVersion other;
+    if (o == null) {
+      return 1;
+    } else if (o instanceof DataprocImageVersion) {
+      other = (DataprocImageVersion) o;
+    } else {
+      other = new DataprocImageVersion(o.toString());
+    }
+
+    // Go over each element of the version and compare.
+    // if elements have different lengths, 0 is taken as the value for each of the missing segments
```

Comment:
this doesn't seem right, since 1.3 translates to the most recent version of 1.3.x

Reply:
For the purposes of the comparison we do in our code, we require "version 1.5 and later" which effectibely means "1.5.0 and later".

so 1.4.x will fail, but [1.5.0, ...) including 2.0.x, etc. will succeed.